esphome/components/mhz14a/sensor.py

```diff
@@ -0,0 +1,47 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import automation
+from esphome.automation import maybe_simple_id
+from esphome.components import sensor, uart
+from esphome.const import CONF_CO2, CONF_ID, CONF_HUMIDITY, ICON_MOLECULE_CO2, \
+    UNIT_PARTS_PER_MILLION, UNIT_PERCENT, ICON_WATER_PERCENT
+
+DEPENDENCIES = ['uart']
+
+CONF_AUTOMATIC_BASELINE_CALIBRATION = 'automatic_baseline_calibration'
+
+mhz14a_ns = cg.esphome_ns.namespace('mhz14a')
+MHZ14AComponent = mhz14a_ns.class_('MHZ14AComponent', cg.PollingComponent, uart.UARTDevice)
+MHZ14ACalibrateZeroAction = mhz14a_ns.class_('MHZ14ACalibrateZeroAction', automation.Action)
+
+CONFIG_SCHEMA = cv.Schema({
+    cv.GenerateID(): cv.declare_id(MHZ14AComponent),
+    cv.Required(CONF_CO2): sensor.sensor_schema(UNIT_PARTS_PER_MILLION, ICON_MOLECULE_CO2, 0),
```

Comment:
You have this set to `Required` but then check if it's in the config. And also have checks in the cpp. Should it be Optional? If not please change the other code sections to assume it will be there.

```suggestion
    cv.Optional(CONF_CO2): sensor.sensor_schema(UNIT_PARTS_PER_MILLION, ICON_MOLECULE_CO2, 0),
```